src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -350,15 +349,15 @@ SHAMapInnerNode::addRaw(Serializer& s, SHANodeFormat format) const
         {
             s.add32 (HashPrefix::innerNode);
 
-            for (int i = 0; i < 16; ++i)
+            for (int i = 0; i < mHashes.size(); ++i)
```

Comment:
Consider:

    for (auto const& h : mHashes)
        s.add256 (h.as_uint256());